source/jormungandr/jormungandr/parking_space_availability/bss/cykleo.py

```diff
@@ -0,0 +1,147 @@
+                    response.status_code))
+                return None
+            return response
+        except pybreaker.CircuitBreakerError as e:
+            logging.getLogger(__name__).error('cykleo service dead (error: {})'.format(e))
+        except requests.Timeout as t:
+            logging.getLogger(__name__).error('cykleo service timeout (error: {})'.format(t))
+        except Exception as e:
+            logging.getLogger(__name__).exception('cykleo error : {}'.format(str(e)))
+        return None
+
+    @cache.memoize(app.config['CACHE_CONFIGURATION'].get('TIMEOUT_CYKLEO_JETON', 10*60))
+    def get_access_token(self):
+        headers = {
+            "Content-Type": "application/json",
+            "Accept": "application/json"
+        }
+        data = {"username": self.username, "password": self.password}
+        if self.service_id is None:
+            data.update({"serviceId": self.service_id})
+
+        response = self.service_caller(method=requests.post, url='{}/pu/auth'.format(self.url),
+                                       headers=headers, data=json.dumps(data))
+        if not response:
+            return None
+        content = response.json()
+        access_token = content.get("access_token")
+        if not access_token:
+            logging.getLogger(__name__).error('cykleo, access_token not exist in response')
+            return None
+        return access_token
+
+    @cache.memoize(app.config['CACHE_CONFIGURATION'].get('TIMEOUT_CYKLEO', 30))
+    def _call_webservice(self):
+        access_token = self.get_access_token()
+        headers = {'Authorization': 'Bearer {}'.format(access_token)}
+        data = self.service_caller(method=requests.get, url='{}/pu/stations/availability'.format(self.url),
+                                   headers=headers)
+        stands = {}
+        if not data:
+            return stands
+        for s in data.json():
+            stands[str(s['station']['assetStation']['commercialNumber'])] = s
+        return stands
+
+    def support_poi(self, poi):
+        properties = poi.get('properties', {})
+        return properties.get('operator', '').lower() in self.operators and \
+               properties.get('network', '').lower() == self.network
+
+    def status(self):
+        return {'network': self.network, 'operators': self.operators}
+
+    def feed_publisher(self):
+        return self._feed_publisher
+
+    def __repr__(self):
+        return ('cykleo-{}'.format(self.network)).encode('UTF-8')
+
+    def get_informations(self, poi):
+        ref = poi.get('properties', {}).get('ref')
+        data = self._call_webservice()
+        if not data:
+            return None
+        station = data.get(ref)
+        if not station:
+            return None
+        if 'availableDockCount' in station \
+                and ('availableClassicBikeCount' in station or 'availableElectricBikeCount' in station):
+            return Stands(station.get('availableDockCount'),
```

Comment:
Don't we also want to check that status is IN_SERVICE also? (poke @kinnou02 )

Reply:
We do not know what to do with the status, for the moment we leave like that